src/plugins/bbsync/sync_thread.cpp

```diff
@@ -66,19 +66,24 @@ void
 BlackBoardSynchronizationThread::init()
 {
 	logger->log_debug(name(), "Initializing");
-	unsigned int check_interval = 0;
+	check_interval_           = 0;
+	connect_failed_since_     = 0;
+	missing_heartbeat_report_ = 0;
+	timewait_                 = 0;
 	try {
 		host_ = config->get_string((peer_cfg_prefix_ + "host").c_str());
 		port_ = config->get_uint((peer_cfg_prefix_ + "port").c_str());
 
-		check_interval = config->get_uint((bbsync_cfg_prefix_ + "check_interval").c_str());
+		check_interval_ = config->get_uint((bbsync_cfg_prefix_ + "check_interval").c_str());
+		missing_heartbeat_report_ =
+		  config->get_uint((bbsync_cfg_prefix_ + "missing_heartbeat_report").c_str());
```

Comment:
if the heartbeat config is not set, then the thrown exception is misleading.
Same goes for the check interval config. Either set them in separate try-catch blocks or use `get_uint_or_default(<path>, <default_value>)` without try catch blocks.

Reply:
I am going with the new try catch, since adding a magic number as default seems to be missing the point of using config files.